sonosco/training/experiment.py

```diff
@@ -99,12 +119,78 @@ def add_directory(self, dir_name):
         # create directory
         path_utils.try_create_directory(dir_path)
 
+    def setup_model_trainer(self,
```

Comment:
to not repeat all the arguments twice I would rather have here `*args, **kwargs` and pass those downwards to ModelTrainer

Reply:
I thought about that aswell, but in my opinion, one should use this function for generally using the model trainer, and when calling this function, can one check what entries you can put there, when it is not specified in this function but just in the model trainer?

Also thought about deleting then the default values in the model trainer init.

Reply:
plus I am not sure how this **kwargs works :D

If I have named arguments in the function, and I pass **kwargs to the model trainer, are then the once that are just for the experiment function left out? or does this lead to an error when there are named arguments, that are not in the model trainer?

Reply:
My biggest argument for why repeating all the arguments is bad is that if you want to change something you always have to remember to change it at three places and this is definitely bad coding style then. The **kwargs to the model trainer would throw an error if you give some arguments which shouldn't be there unless ModelTrainer also has **kwargs in the end which picks up all the unmatched args.